ggshield/auth.py

```diff
@@ -10,21 +13,65 @@
     type=click.Choice(["token"]),
     help="Authentication method.",
 )
+@click.option(
+    "--instance",
+    required=False,
+    type=str,
+    help="URL of the instance to authenticate to.",
+)
 @click.pass_context
-def login_cmd(ctx: click.Context, method: str) -> int:
+def login_cmd(ctx: click.Context, method: str, instance: str) -> int:
     """Authenticate to your GitGuardian account."""
+    config = ctx.obj["config"]
+
+    if instance:
+        if not instance.startswith("https://"):
+            raise click.BadParameter("Instance must be a valid https URL.")
+
+        # Normalize path by removing the path, if it is set
+        parsed_url = urlparse(instance)
+        instance = f"https://{parsed_url.netloc}"
+
+    if not instance:
+        instance = config.auth_config.default_instance
+
     if method == "token":
         token = click.prompt("Enter your GitGuardian API token", hide_input=True)
         if not token:
             raise click.ClickException("No API token was provided.")
 
-        config = ctx.obj["config"]
+        config.auth_config.current_instance = instance
```

Comment:
nit: this will be used even it the method is not "token", this can go outside of the if block (before)